File: mobile/lib/core/network/interceptors/accept_language_interceptor.dart
import 'package:dio/dio.dart';
import 'package:vaxiil_mobile/core/locale/locale_manager.dart';

/// Sends the active UI locale so Django LocaleMiddleware can translate
/// choice titles and API error messages.
class AcceptLanguageInterceptor extends Interceptor {
  AcceptLanguageInterceptor([LocaleManager? localeManager])
      : _localeManager = localeManager ?? LocaleManager();

  final LocaleManager _localeManager;

  @override
  void onRequest(RequestOptions options, RequestInterceptorHandler handler) {
    options.headers['Accept-Language'] = _localeManager.languageCode;
    handler.next(options);
  }
}

File: web/src/app/core/i18n/accept-language.interceptor.ts
import { HttpInterceptorFn } from '@angular/common/http';
import { inject } from '@angular/core';

import { LocaleService } from '@/core/i18n/locale.service';

/** Attach Accept-Language so Django LocaleMiddleware can translate API copy. */
export const acceptLanguageInterceptor: HttpInterceptorFn = (req, next) => {
  const locale = inject(LocaleService);
  if (req.url.includes('/assets/i18n/')) {
    return next(req);
  }
  return next(
    req.clone({
      setHeaders: { 'Accept-Language': locale.acceptLanguage() },
    }),
  );
};
